clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java

```diff
@@ -222,7 +222,14 @@
     private static final String EXCLUDE_INTERNAL_TOPICS_DOC = "Whether records from internal topics (such as offsets) should be exposed to the consumer. "
                                                             + "If set to <code>true</code> the only way to receive records from an internal topic is subscribing to it.";
     public static final boolean DEFAULT_EXCLUDE_INTERNAL_TOPICS = true;
-    
+
+    /**
+     * <code>leave.group.on.close</code>
+     * Whether or not the consumer should leave the group on close. If set to <code>false</code> then a rebalance
+     * won't occur until <code>session.timeout.ms</code> expires.
+     */
+    static final String LEAVE_GROUP_ON_CLOSE_CONFIG = "leave.group.on.close";
```

Comment:
Can you explain the reasoning for making this an internal config? If it's useful for Streams, won't it also be useful for other libraries that try to do similar things?